### PR TITLE
Modifications to docs

### PR DESCRIPTION
**Pull request by M00nF1sh**
1. Adding new architecture digram(including IP mode)
2. Adding rough introduction for IP mode
3. Fix broken links in doc

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -49,12 +49,15 @@ Along with the above, the controller also...
   recover if the controller were to be restarted.
 
 ### Ingress Traffic
-
-This section details how traffic reaches the cluster.
-
-As seen above, the ingress traffic for controller-managed resources starts at the ALB and reaches the Kubernetes nodes through each service's NodePort. This means that
-services referenced from ingress resource must be exposed on a node port in order to be
-reached by the ALB.
+ALB Ingress controller supports two traffic mode:
```
Comment (bigkraig) [inline docs/README.md]: s/mode/modes/

File: docs/README.md [hunk 2]
```diff
@@ -49,12 +49,15 @@ Along with the above, the controller also...
...
+* Instance mode
+* IP mode
+
+By default, `Instance mode` is used, users can explicitly select mode via `alb.ingress.kubernetes.io/target-type` annotation.
```
Comment (bigkraig) [inline docs/README.md]: s/select mode/select the mode/

File: docs/README.md [hunk 3]
```diff
@@ -49,12 +49,15 @@ Along with the above, the controller also...
...
+#### Instance mode
+Ingress traffic starts at the ALB and reaches the Kubernetes nodes through each service's NodePort. This means that services referenced from ingress resources must be exposed by `type:NodePort` in order to be reached by the ALB.
+#### IP mode
+Ingress traffic starts at the ALB and reaches the Kubernetes pods directly. CNIs must supports directly accessible POD ip via [secondary IP addresses on ENI](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-eni.html).
```
Comment (bigkraig) [inline docs/README.md]: s/supports/support/

File: docs/guide/setup.md [hunk 4]
```diff
@@ -24,7 +24,7 @@ The controller determines subnets to deploy each ALB to based on an annotation o
 
 ##### Via annotation
 
-`alb.ingress.kubernetes.io/subnets` may be specified in each ingress resource with the subnet IDs or `Name` tags. This allows for flexibility in where ALBs land. This list of subnets must include 2 or more that exist in unique availability zones. See the [annotations documentation](ingress-resources.md#annotations) for more details.
+`alb.ingress.kubernetes.io/subnets` may be specified in each ingress resource with the subnet IDs or `Name` tags. This allows for flexibility in where ALBs land. This list of subnets must include 2 or more that exist in unique availability zones. See the [annotations documentation](../api/ingress.md#annotations) for more details.
```
Comment (bigkraig) [inline docs/guide/setup.md]: s/This list of subnets/The list of subnets/

Comment (M00nF1sh) [reply docs/guide/setup.md]: fixed 😄

File: docs/README.md (hunk 3 above)
Comment (M00nF1sh) [reply docs/README.md]: fixed 😄

File: docs/README.md (hunk 2 above)
Comment (M00nF1sh) [reply docs/README.md]: fixed 😄

File: docs/README.md (hunk 1 above)
Comment (M00nF1sh) [reply docs/README.md]: fixed 😄

**Pull request merged**